src/test/ui/type-inference/sort_by_key.stderr

```diff
@@ -0,0 +1,11 @@
+error[E0282]: type annotations needed
+  --> $DIR/sort_by_key.rs:3:9
+   |
+LL |     lst.sort_by_key(|&(v, _)| v.iter().sum());
+   |         ^^^^^^^^^^^                    --- help: consider specifying the type argument in the method call: `sum::<S>`
+   |         |
+   |         cannot infer type for `K`
```

Comment:
This is pre-existing, but I wonder if we could say 

```
LL |     lst.sort_by_key(|&(v, _)| v.iter().sum());
   |         ^^^^^^^^^^^                    --- help: consider specifying the type argument in the method call: `sum::<S>`
   |         |
   |         cannot infer type for type parameter `K` declared on `sort_by_key`
```

or something like that?

Reply:
I know, it's kind of tricky to do it right at the moment, but the slight rewording could be tackled (mentioning sort_by_key might be much harder)